syntax = "proto3";

package testgrpc;

service Tienda {
  rpc AgregarTienda(AgregarTiendaRequest) returns (AgregarTiendaResponse) {}
  rpc ModificarTienda(ModificarTiendaRequest) returns (ModificarTiendaResponse) {}
  rpc EliminarTienda(EliminarTiendaRequest) returns (EliminarTiendaResponse) {}
  rpc TraerTodasLasTiendas(TraerTodasLasTiendasRequest) returns (TraerTodasLasTiendasResponse) {}
  rpc TraerTodasLasTiendasPorEstado(TraerTodasLasTiendasPorEstadoRequest) returns (TraerTodasLasTiendasPorEstadoResponse) {}
  rpc TraerTiendaPorCodigo(TraerTiendaPorCodigoRequest) returns (TraerTiendaPorCodigoResponse) {}
}

message TiendaGrpcDTO {
  int64 id = 1;
  string direccion = 2;
  string ciudad = 3;
  string provincia = 4;
  bool habilitado = 5;
}

message AgregarTiendaRequest {
  TiendaGrpcDTO tiendaGrpcDTO = 1;
}

message AgregarTiendaResponse {
    TiendaGrpcDTO tiendaGrpcDTO = 1;
}

message TiendaList {
  repeated TiendaGrpcDTO tiendas = 1;
}

message ModificarTiendaRequest {
  TiendaGrpcDTO tiendaGrpcDTO = 1;
}

message ModificarTiendaResponse {
    TiendaGrpcDTO tiendaGrpcDTO = 1;
}

message EliminarTiendaRequest {
  int64 id = 1;
}

message EliminarTiendaResponse {
    int64 id = 1;
}

message TraerTodasLasTiendasRequest {
}

message TraerTodasLasTiendasResponse {
    TiendaList tiendaList = 1;
}

message TraerTodasLasTiendasPorEstadoRequest {
  bool estado = 1;
}

message TraerTodasLasTiendasPorEstadoResponse {
  TiendaList tiendaList = 1;
}

message TraerTiendaPorCodigoRequest {
  int64 codigo = 1;
}

message TraerTiendaPorCodigoResponse {
    TiendaGrpcDTO tiendaGrpcDTO = 1;
}